src/test/javascript/spec/component/course/course-overview.component.spec.ts

```diff
@@ -645,4 +652,35 @@ describe('CourseOverviewComponent', () => {
 
         expect(component.dropdownOpen).toBeTrue();
     });
+
+    it('should initialize courses attribute when page is loaded', () => {
+        const findAllForDashboardSpy = jest.spyOn(courseService, 'findAllForDashboard');
+        findAllForDashboardSpy.mockReturnValue(of(new HttpResponse({ body: coursesDashboard, headers: new HttpHeaders() })));
+
+        component.ngOnInit();
+
+        expect(component.courses).toEqual(courses);
+        expect(component.courses?.length).toBe(1);
+    });
+
+    it('should set dropdownCourses to true when clicking first time', () => {
+        component.dropdownCourses = false;
+        fixture.detectChanges();
+
+        const clickOnDropdownButton = fixture.nativeElement.querySelector('#courseIconCustom');
+        clickOnDropdownButton.click();
+
+        expect(component.dropdownCourses).toBeTrue();
+    });
+
+    it('should set dropdownCourses to false when reclicking', () => {
+        component.dropdownCourses = false;
+        fixture.detectChanges();
+
+        const clickOnDropdownButton = fixture.nativeElement.querySelector('#courseIconCustom');
+        clickOnDropdownButton.click();
+        clickOnDropdownButton.click();
+
+        expect(component.dropdownCourses).toBeFalse();
+    });
```

Comment:
I think it would make sense to add a test for the component to display the correct courses, e.g. "should not display the currently selected course in dropdown"

Reply:
done 👍